tests/regression/src/lib.rs

```diff
@@ -55,24 +79,228 @@ mod tests {
             cg::cachegrind::start_instrumentation();
         }
     }
-    /// Environment variable to determine whether to run under valgrind or solely test functionality.
-    fn is_running_under_valgrind() -> bool {
-        env::var("ENABLE_VALGRIND").is_ok()
+
+    #[derive(Debug)]
+    enum RegressionTestMode {
+        Default,
+        Valgrind,
+        Diff,
+    }
+
+    impl RegressionTestMode {
+        fn from_env() -> Self {
+            if env::var("ENABLE_VALGRIND").is_ok() {
+                RegressionTestMode::Valgrind
+            } else if env::var("DIFF_MODE").is_ok() {
+                RegressionTestMode::Diff
+            } else {
+                RegressionTestMode::Default
+            }
```

Comment:
I wonder if it would be cleaner to have a single environment variable for this, especially where they're mutually exclusive? Something like `PERF_MODE={valgrind,diff}`

Reply:
Yea, that is neater and easier for the user (+ better for future extensibility if we want to try and make everything run with one command w/ dyn linking someday). I'll update the configuration and README so we only use one environment variable.